signer/src/bitcoin/validation.rs

```diff
@@ -1088,10 +1130,19 @@ mod tests {
             witness: Witness::new(),
         });
 
-        let status =
-            mapping
-                .report
-                .validate(mapping.chain_tip_height, &tx, TX_FEE, Amount::MAX_MONEY);
+        let status = mapping.report.validate(
+            mapping.chain_tip_height,
+            &tx,
+            TX_FEE,
+            &SbtcLimits::new(
+                None,
+                Some(Amount::from_sat(per_deposit_minimum)),
+                Some(Amount::from_sat(per_deposit_cap)),
+                None,
+                None,
+                None,
+            ),
+        );
```

Comment:
nit: it's much easier to read if we just split this out like:
```rust
let limits = SbtcLimits::new(
    None,
    Some(Amount::from_sat(per_deposit_minimum)),
    Some(Amount::from_sat(per_deposit_cap)),
    None,
    None,
    None,
);
let status = mapping
    .report
    .validate(mapping.chain_tip_height, &tx, TX_FEE, limits);
```
or if we add a test helper implementation, we could do:
```rust
#[test_case(DepositReportErrorMapping {
    report: DepositRequestReport { ... },
    status: InputValidationResult::AmountTooLow,
    chain_tip_height: 2,
    sbtc_limits: SbtcLimits::per_deposit(100_000_000, u64::MAX),
} ; "amount-too-low")]
fn deposit_report_validation(mapping: DepositReportErrorMapping) {
    ...
    let status = mapping
        .report
        .validate(mapping.chain_tip_height, &tx, TX_FEE, &mapping.sbtc_limits:);
```

Reply:
now the limits are included in the DepositReportErrorMapping  https://github.com/stacks-network/sbtc/pull/1124/commits/b813d2062d178cb5b961b065585c12d0ea04bb0b